### PR TITLE
 Add `IgnoredMethods` option to `Style/NumericPredicate` cop

### PR DESCRIPTION
**Pull request by AlexWayfer**
Allow to ignore ORM (Sequel) methods.

Also take out `IgnoredMethods` mixin.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/mixin/ignored_methods.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # This module encapsulates the ability to ignore certain methods when
+    # parsing.
+    module IgnoredMethods
```
Comment (rrosenblum) [inline lib/rubocop/cop/mixin/ignored_methods.rb]: Good call on making this a reusable module.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,10 +2,16 @@
...
 
+### New features
+
+* [#6148](https://github.com/bbatsov/rubocop/pull/6148): Add `IgnoredMethods` option to `Style/NumericPredicate` cop. ([@AlexWayfer][])
+
 ### Changes
 
 * [#6137](https://github.com/rubocop-hq/rubocop/pull/6137): Allow `db` to allowed names of `Naming/UncommunicativeMethodParamName` cop in default config. ([@mkenyon][])
 
+* [#6148](https://github.com/bbatsov/rubocop/pull/6148): Add `IgnoredMethods` mixin. ([@AlexWayfer][])
```
Comment (rrosenblum) [inline CHANGELOG.md]: We don't normally create changelog entries for internal changes.

Comment (AlexWayfer) [reply CHANGELOG.md]: OK, sorry, removed.

File: lib/rubocop/cop/mixin/ignored_methods.rb (hunk 1 above)
Comment (AlexWayfer) [reply lib/rubocop/cop/mixin/ignored_methods.rb]: Thanks. I was surprised at the number of ways to check the same config in different cops (with `.to_a` or without, `map(&:to_sym)` or `name.to_s`).

**Comment by bbatsov** [comment]
Nicely done!

**Comment by AlexWayfer** [comment]
Thanks!

**Pull request merged**